arviz/plots/backends/matplotlib/traceplot.py

```diff
@@ -192,6 +192,8 @@ def plot_trace(
     fill_kwargs = matplotlib_kwarg_dealiaser(fill_kwargs, "fill_between")
     rug_kwargs = matplotlib_kwarg_dealiaser(rug_kwargs, "scatter")
     rank_kwargs = matplotlib_kwarg_dealiaser(rank_kwargs, "bar")
+    if compact:
+        rank_kwargs.setdefault("bar_kwargs", {"alpha": 0.2})
```

Comment:
I think this will only work in a small subset of cases and it should be something like:

```suggestion
    if compact:
        rank_kwargs.setdefault("bar_kwargs", {})
        rank_kwargs["bar_kwargs"].setdefault("alpha", 0.2)
```